src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,233 @@
+						&& accident.getAccidentId().equals(index)) {
+					roadAccident = accident;
+					break;
+				}
+			}
+		}
+		return roadAccident;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		Collection<RoadAccident> colls = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			colls = new ArrayList<RoadAccident>();
+			for (RoadAccident accident : roadAccidentList) {
+				if (accident.getLongitude() <= maxLongitude
+						&& accident.getLongitude() >= minLongitude
+						&& accident.getLatitude() <= maxLatitude
+						&& accident.getLatitude() >= minLatitude) {
+					colls.add(accident);
+				}
+			}
+		}
+		return colls;
+	}
+
+	/**
+	 * count incidents by road surface conditions ex: wet -> 2 dry -> 5
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition7() {
+		//roadSurfaceConditions
+		Map<String, Long> resultMap = null;
+		
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			resultMap = new HashMap<String, Long>(); 
+			Multimap<String, RoadAccident> multiMap = ArrayListMultimap.create();
+			for (RoadAccident accident : roadAccidentList) {
+				multiMap.put(accident.getRoadSurfaceConditions(),
+						accident);				
+			}			
+			for(String key:multiMap.keySet()){
+				resultMap.put(key, new Long(multiMap.get(key).size()));
+			}
+		}
+		
+		return resultMap;
+		
+		
+		
+	}
+
+	/**
+	 * find the weather conditions which caused the top 3 number of incidents as
+	 * example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1
+	 * in foggy, then your result list should contain {rain, sunny, snow} - top
+	 * three in decreasing order
+	 * 
+	 * @return
+	 */
+	public List<String> getTopThreeWeatherCondition7() {
+		List<String> result = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			result = new ArrayList<String>();
+			Map<String, Long> counterMap = new HashMap<String, Long>();			//
+			Multimap<String, RoadAccident> multiMap = ArrayListMultimap.create();
+			for (RoadAccident accident : roadAccidentList) {
+				multiMap.put(accident.getWeatherConditions(),
+						accident);				
+			}
+			
+			for(String key:multiMap.keySet()){
+				counterMap.put(key, new Long(multiMap.get(key).size()));
+			}
+			
+			List<Map.Entry<String, Long>> entryList = new ArrayList<Map.Entry<String, Long>>();
+			
+			for(Entry<String, Long> entry:counterMap.entrySet()){
+				entryList.add(entry);				
+			}
+			Collections.sort(entryList,new Comparator<Map.Entry<String,Long>>(){
+				@Override
+				public int compare(Entry<String, Long> o1,
+						Entry<String, Long> o2) {						
+					return o1.getValue().compareTo(o2.getValue());
```

Comment:
This will sort in ascending order, so you will need to get three last elements. 

I think it's more natural to sort in a descending order and get three first elements. What do you think?